src/dialer/index.js

```diff
@@ -190,6 +202,48 @@ class Dialer {
     log('token %d released', token)
     this.tokens.push(token)
   }
+
+  /**
+   * Resolve multiaddr recursively.
+   *
+   * @param {Multiaddr} ma
+   * @returns {Promise<Array<Multiaddr>>}
+   */
+  async _resolve (ma) {
```

Comment:
Once we support recursion in `multiaddr`, as well as `dns4`+`dns6` resolution, we will not need this function, only the function below